messenger.go

```diff
@@ -51,9 +54,13 @@ type Messenger struct {
 }
 
 // New creates a new Messenger. You pass in Options in order to affect settings.
-func New(mo Options) *Messenger {
+func New(mo Options, mux *http.ServeMux) *Messenger {
```

Comment:
I don't really want to change the function signature of this function.

I'd prefer it if we added this onto the `Options` struct.